sixtyfps_runtime/interpreter/eval.rs

```diff
@@ -477,7 +477,25 @@ pub fn eval_expression(expression: &Expression, local_context: &mut EvalLocalCon
                 sixtyfps_compilerlib::expression_tree::ImageReference::AbsolutePath(path) => {
                     corelib::graphics::Image::load_from_path(std::path::Path::new(path))
                 }
-                sixtyfps_compilerlib::expression_tree::ImageReference::EmbeddedData { .. } => panic!("Resource embedding is not supported by the interpreter")
+                sixtyfps_compilerlib::expression_tree::ImageReference::EmbeddedData { resource_id, extension } => {
+                    let toplevel_instance = match local_context.component_instance {
+                        ComponentInstance::InstanceRef(instance) => instance.toplevel_instance(),
+                        ComponentInstance::GlobalComponent(_) => unimplemented!(),
+                    };
+                    let extra_data = toplevel_instance.component_type.extra_data_offset.apply(toplevel_instance.as_ref());
+                    let path = extra_data.embedded_file_resources.get(resource_id).expect("internal error: invalid resource id");
+
+                    let data = match path.strip_prefix("builtin:/") {
+                        None => unimplemented!(),
+                        Some(builtin) => {
+                         sixtyfps_compilerlib::library::load_file(std::path::Path::new(builtin))
+                                .expect("non-existent internal file referenced")
+                        }
+                    };
+                    Ok(corelib::graphics::Image::from(
+                        corelib::graphics::ImageInner::EmbeddedData{ data: corelib::slice::Slice::from_slice(data), format: corelib::slice::Slice::from_slice(interned_str(extension).as_bytes()) }
```

Comment:
Ah, i see where interned_str is used now. Somehow the github interface was hiding it from me.

Do we really need to support embedding data in the interpreter for generic data? I believe this is enough to do it for the data coming from the library, for which the data is static.

Reply:
Reading more, i see that the problem is the extension.
Could `load_file` also return the filename or extension as a &'static  (since the builtin library has the filenames built in)

Reply:
You’re right, that’s an elegant solution.